### PR TITLE
Set version to 1.0.0

### PR DESCRIPTION
**Pull request by mortenpi**
_Not meant to be just merged yet, opening this as a PSA._

The 1.0.0 release will contain quite a few breaking changes, so it would be good to get some feedback from the community. It would be amazing if people could play with `Documenter #master` for your own docs a bit (just `pkg> add Documenter#master` to `docs/Project.toml`), before we foist it on everyone with a release.

In particular, it would be good to know how painful upgrading to 1.0 is, and if there is anything that should be added to the release notes (including on how to upgrade). But also, any feedback or thoughts are very welcome.

* You can find the release notes here: https://documenter.juliadocs.org/dev/release-notes/#Unreleased
* Note that there are also a few outstanding issues [on the 1.0.0 milestone](https://github.com/JuliaDocs/Documenter.jl/milestone/107).

### CODE REVIEW

**Comment by odow** [comment]
The JuMP docs are building on a very recent commit (https://github.com/JuliaDocs/Documenter.jl/tree/e2bd1940ad9a8ad1320252a2fc3b4468b770e3a4), so I'm happy.

I understand the motivation for the page size limit, but I still found it quite annoying.

**Comment by mortenpi** [comment]
@odow What is the main source of the large pages in the JuMP manual? Is figured that we could fix with https://github.com/JuliaDocs/Documenter.jl/issues/2143?

**Comment by odow** [comment]
It's both. We have some very long pages, and some pages with lots of figures and data tables.

File: CHANGELOG.md [hunk 1]
```diff
@@ -3,7 +3,7 @@
 The format is based on [Keep a Changelog](https://keepachangelog.com/en/1.0.0/),
 and this project adheres to [Semantic Versioning](https://semver.org/spec/v2.0.0.html).
 
-## Unreleased
+## Version [v1.0.0] - 2023-08-??
```
Comment (mortenpi) [inline CHANGELOG.md:6]:
```suggestion
## Version [v1.0.0] - 2023-09-15
```

**Pull request merged**